src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/HeadersAuthenticationFilter.java

```diff
@@ -82,6 +86,14 @@ protected void authenticate(HttpServletRequest req) {
                 }
                 
             }
+            if (!everyoneFound && addEveryOneGroup) {
```

Comment:
I think this is because headers may not contain the everyone group but the geostore model require it to share public resources, isn't it?
Does it work even for guest user?

Can you please explain a little bit this in a comment